test/rubygems/test_gem_commands_push_command.rb

```diff
@@ -250,6 +251,45 @@ def test_sending_gem_to_disallowed_push_host
     assert_match response, @ui.error
   end
 
+  def test_sending_gem_defaulting_to_allowed_push_host
+    host = "http://privategemserver.example"
+
+    @spec, @path = util_gem "freebird", "1.0.1" do |spec|
+      spec.metadata.delete('default_gem_server')
+      spec.metadata['allowed_push_host'] = host
+    end
+
+    api_key = "PRIVKEY"
+
+    keys = {
+      host => api_key
+    }
+
+    FileUtils.mkdir_p File.dirname Gem.configuration.credentials_path
+    open Gem.configuration.credentials_path, 'w' do |f|
+      f.write keys.to_yaml
+    end
+    Gem.configuration.load_api_keys
+
+    FileUtils.rm Gem.configuration.credentials_path
```

Comment:
yeah they are repeated several times.  it's possible we can use the `set_api_key` function in #1485